kairon/chat/handlers/channels/voice.py

```
@@ -90,40 +98,91 @@ async def handle_incoming_call(self) -> Text:
+            return provider_impl.build_hangup_response(messages)
 
         return provider_impl.build_voice_response(messages, config["call_url"])
 
+    async def _has_prior_conversation(self, sender_id: Text) -> bool:
+        """Check ChannelLogs for a welcome_sent entry — true means this call was already greeted."""
+        try:
+            return ChannelLogs.objects(
+                bot=self.bot, message_id=sender_id,
+                type=ChannelTypes.VOICE.value, status="welcome_sent"
+            ).count() > 0
+        except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)